File: backends/nest/src/common/exceptions/capsule-not-found.exception.ts
// 404 胶囊不存在异常 — 查询或删除时 code 未匹配到记录
import { HttpStatus } from '@nestjs/common';
import { AppException } from './app.exception';

export class CapsuleNotFoundException extends AppException {
  constructor(code: string) {
    super('CAPSULE_NOT_FOUND', `胶囊不存在：${code}`, HttpStatus.NOT_FOUND);
  }
}

File: fullstacks/spring-boot-mvc/src/main/java/com/hellotime/exception/CapsuleNotFoundException.java
package com.hellotime.exception;

/**
 * 胶囊未找到异常
 * 当查询的胶囊码不存在时抛出此异常
 * 会被 GlobalExceptionHandler 捕获并返回 404 响应
 */
public class CapsuleNotFoundException extends RuntimeException {

    /**
     * 构造方法
     *
     * @param code 未找到的 8 位胶囊码
     */
    public CapsuleNotFoundException(String code) {
        super("胶囊不存在：" + code);
    }
}
